### ISSUE TITLE
Fix backpropagation through time

### ISSUE THREAD

**Issue by dbeinhauer** [dbeinhauer/mcs-source#7]
Backpropagation through time does not work correctly in the current implementation.

Possible problems might be related to reassignment of the hidden states for each visible time step during training.
- we will need to figure out how to perform this correctly in pytorch 

- [x] #34

**Comment by dbeinhauer** [comment]
Currently, it looks that the correct way how to perform this is to simply run optimizer step before each hidden state reset with targets.

So, let the network do the steps as before. Calculate loss and do backward and optimizer steps after each visible time step prediction. This should secure correct backpropagation through time (there is not any artificial state change for the selected time sequence).
